simGanESS/source/geometries/GaP.cc

```diff
@@ -332,49 +449,173 @@ void GaP::BuildTPC(G4Material* gas, G4Material* mesh_mat, G4LogicalVolume* logic
+    G4VSolid *solidSub_enclosure_pmt3  = new G4SubtractionSolid("EnclosurePMT", solidSub_enclosure_pmt2, solid_pmt, 0,G4ThreeVector(0.,          0.,        relative_pmt_z));
+    G4VSolid *solidSub_enclosure_pmt4  = new G4SubtractionSolid("EnclosurePMT", solidSub_enclosure_pmt3, solid_pmt, 0,G4ThreeVector(36.253*mm,   2.949*mm,  relative_pmt_z));
+    G4VSolid *solidSub_enclosure_pmt5  = new G4SubtractionSolid("EnclosurePMT", solidSub_enclosure_pmt4, solid_pmt, 0,G4ThreeVector(-20.68*mm,   29.922*mm, relative_pmt_z));
+    G4VSolid *solidSub_enclosure_pmt6  = new G4SubtractionSolid("EnclosurePMT", solidSub_enclosure_pmt5, solid_pmt, 0,G4ThreeVector(15.573*mm,   32.871*mm, relative_pmt_z));
+
+    G4LogicalVolume *logic_enclosure_pmt = new G4LogicalVolume(solidSub_enclosure_pmt6, steel, "EnclosurePMT");
+
+    new G4PVPlacement(0, G4ThreeVector(0, 0, -enclosure_pmt_z), logic_enclosure_pmt, "EnclosurePMT", logic_vessel, false, 0, true);
+
+    // Vacuum inside the pmt enclosure
+    G4double enclosurevac_pmt_z = vessel_length_/2 - enclosurevac_pmt_length_/2;
+    G4double relativevac_pmt_z  = enclosurevac_pmt_z - pmt_z;
+
+    G4Tubs *solid_enclosurevac_pmt = new G4Tubs("EnclosureVacPMT", 0, enclosure_pmt_rad_, (enclosurevac_pmt_length_)/2, 0., 360.*deg);
+    G4VSolid *solidSub_enclosurevac_pmt0  = new G4SubtractionSolid("EnclosureVacPMT", solid_enclosurevac_pmt, solid_pmt,     0,G4ThreeVector(-15.573*mm, -32.871*mm, relativevac_pmt_z));
+    G4VSolid *solidSub_enclosurevac_pmt1  = new G4SubtractionSolid("EnclosureVacPMT", solidSub_enclosurevac_pmt0, solid_pmt, 0,G4ThreeVector(20.68*mm,   -29.922*mm, relativevac_pmt_z));
+    G4VSolid *solidSub_enclosurevac_pmt2  = new G4SubtractionSolid("EnclosureVacPMT", solidSub_enclosurevac_pmt1, solid_pmt, 0,G4ThreeVector(-36.253*mm, -2.949*mm,  relativevac_pmt_z));
+    G4VSolid *solidSub_enclosurevac_pmt3  = new G4SubtractionSolid("EnclosureVacPMT", solidSub_enclosurevac_pmt2, solid_pmt, 0,G4ThreeVector(0.,          0.,        relativevac_pmt_z));
+    G4VSolid *solidSub_enclosurevac_pmt4  = new G4SubtractionSolid("EnclosureVacPMT", solidSub_enclosurevac_pmt3, solid_pmt, 0,G4ThreeVector(36.253*mm,   2.949*mm,  relativevac_pmt_z));
+    G4VSolid *solidSub_enclosurevac_pmt5  = new G4SubtractionSolid("EnclosureVacPMT", solidSub_enclosurevac_pmt4, solid_pmt, 0,G4ThreeVector(-20.68*mm,   29.922*mm, relativevac_pmt_z));
+    G4VSolid *solidSub_enclosurevac_pmt6  = new G4SubtractionSolid("EnclosureVacPMT", solidSub_enclosurevac_pmt5, solid_pmt, 0,G4ThreeVector(15.573*mm,   32.871*mm, relativevac_pmt_z));
+
+    G4LogicalVolume *logic_enclosurevac_pmt = new G4LogicalVolume(solidSub_enclosurevac_pmt6, vacuum, "EnclosureVacPMT");
+
+    new G4PVPlacement(0, G4ThreeVector(0., 0, enclosure_pmt_z - enclosurevac_pmt_z), logic_enclosurevac_pmt, "EnclosureVacPMT", logic_enclosure_pmt, false, 0, true);
```

Comment:
Are you sure about the Z position here? If this were a daughter of logic_vessel it would make sense I think, but its mother is the logic_enclosure_pmt. I'm not saying is wrong, but using the global position of the mother to position a daughter inside the mother looks suspicious (it may work but just because the centering is as is).